File: framework/languages/java/samples/java/TicTacToe.SessionGateway/src/main/java/systems/zlink/samples/tictactoe/sessiongateway/shared/actors/PlayerActor.java
package systems.zlink.samples.tictactoe.sessiongateway.shared.actors;

import java.util.List;
import systems.zlink.framework.actors.ZLinkActor;
import systems.zlink.framework.actors.ZLinkActorContext;

public final class PlayerActor implements ZLinkActor {
    private final String actorId;
    private final RecordingActorContext context = new RecordingActorContext();

    public PlayerActor(String actorId) {
        this.actorId = actorId;
    }

    @Override
    public String actorId() {
        return actorId;
    }

    @Override
    public ZLinkActorContext context() {
        return context;
    }

    public List<String> pushes() {
        return context.pushes();
    }
}

File: framework/languages/java/samples/kotlin/TicTacToe.SessionGateway/src/main/kotlin/systems/zlink/samples/kotlin/tictactoe/sessiongateway/shared/actors/PlayerActor.kt
package systems.zlink.samples.kotlin.tictactoe.sessiongateway.shared.actors

import systems.zlink.framework.actors.ZLinkActor
import systems.zlink.framework.actors.ZLinkActorContext

class PlayerActor(
    private val actorId: String,
) : ZLinkActor {
    private val context = RecordingActorContext()

    override fun actorId(): String = actorId
    override fun context(): ZLinkActorContext = context
    fun pushes(): List<String> = context.pushes()
}
